addon/components/polaris-text-field.js

```diff
@@ -392,79 +389,82 @@ export default Component.extend({
     }
 
     return describedBy.join(' ');
-  }),
+  }).readOnly(),
 
   ariaLabelledBy: computed('id', function() {
     return `${ this.get('id') }Label`;
-  }),
+  }).readOnly(),
 
-  ariaInvalid: computed('error', function() {
-    return !!this.get('error');
-  }),
+  ariaInvalid: bool('error').readOnly(),
 
   inputType: computed('type', function() {
     let type = this.get('type');
 
     return type === 'currency' ? 'text' : type;
-  }),
-
-  input: computed('multiline', function() {
-    let multiline = this.get('multiline');
-
-    return multiline ? 'textarea' : 'input';
-  }),
+  }).readOnly(),
 
   minimumLines: computed('multiline', function() {
     let multiline = this.get('multiline');
 
-    return (typeof multiline === 'number') ? multiline : 1;
-  }),
+    return (typeOf(multiline) === 'number') ? multiline : 1;
+  }).readOnly(),
 
   heightStyle: computed('height', 'multiline', function() {
     let { height, multiline } = this.getProperties('height', 'multiline');
 
-    return (height && multiline) ? htmlSafe(`height: ${ height }px`) : '';
-  }),
+    return (height && multiline) ? htmlSafe(`height: ${ height }px`) : null;
+  }).readOnly(),
+
+  shouldShowSpinner: computed('disabled', 'type', function() {
+    let { type, disabled } = this.getProperties('type', 'disabled');
+
+    return (typeOf(type) === 'number') && !disabled;
+  }).readOnly(),
+
+  addValueListener() {
+    this.addEventListener('keyup', () => {
```

Comment:
Can you check that adding an event handler using an arrow function works in terms of tearing the handler down? I'd think `ember-lifeline` is intelligent enough to handle it properly but worth checking given that this is a leaky pattern for regular adding/removing event handlers...

Reply:
I think we're safe, the example from their docs shows:

```javascript
didInsertElement() {
    this._super(...arguments);
    addEventListener(this, window, 'scroll', (e) => {
      this.set('windowScrollOffset', e.clientY);
    });
  },
```

The only difference is that I'm using the Mixin to create mine which uses the same `addEventListener` method above.